lib/autoBlock.js

```diff
@@ -225,17 +320,29 @@ function runStepFunc (results, callback, step, config) {
   var func = step.func
   var start = Date.now()
 
-  config.onStartStep(step.name, config.data)
+  config.onStartStep({
+    step: {
+      name: step.name
+    },
+    data: config.data,
+    options: config.options,
+    results: results
+  })
 
   var args = utils.mappingArgs(results, step.with, step.withDefaults)
 
   var cb = (err, result) => {
-    var stepOutput = {
-      error: !!err,
+    config.onFinishStep({
+      step: {
+        name: step.name
+      },
+      data: config.data,
       result: !!result,
-      timer: (Date.now() - start) / 1000
-    }
-    config.onFinishStep(step.name, config.data, stepOutput)
+      results: results,
+      error: !!err,
```

Comment:
When we rev a major version number we can upgrade this to the full error object. For now, this functionality has been locked in.

(The short answer is that I was concerned about sensitive information appearing in logs and just mushed it into a boolean. But that shouldn't be the responsibility of _this_ code so it was the wrong thing to do.)